dll/cpl/usrmgr/users.c

```diff
@@ -672,6 +718,43 @@ UsersPageProc(HWND hwndDlg,
             pUserData->hPopupMenu = LoadMenu(hApplet, MAKEINTRESOURCE(IDM_POPUP_USER));
 
             OnInitDialog(hwndDlg);
+			
+            HKEY hKeyInit;
+            LONG lResultInit;
+            lResultInit = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                                        L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+                                        0, KEY_READ, &hKeyInit);
+										
+            if (lResultInit == ERROR_SUCCESS)
+            {
+                  WCHAR szAutoAdminLogonValueInit[2];
+                  DWORD dwTypeInit;
+                  DWORD dwSizeInit = sizeof(szAutoAdminLogonValueInit);
+
+                  lResultInit = RegQueryValueExW(hKeyInit,
+                                                 L"AutoAdminLogon",
+                                                 NULL,
+                                                 &dwTypeInit,
+                                                 (LPBYTE)szAutoAdminLogonValueInit,
+                                                 &dwSizeInit);
+
+                 BOOL bRequireLogonInit = TRUE;
+                 if (lResultInit == ERROR_SUCCESS && dwTypeInit == REG_SZ)
+                 {
+                    if (wcscmp(szAutoAdminLogonValueInit, L"1") == 0)
+                    {
+                       bRequireLogonInit = FALSE;
+                    }
+                 }
```

Comment:
```suggestion
                 if (   lResultInit == ERROR_SUCCESS
                     && dwTypeInit == REG_SZ
                     && wcscmp(szAutoAdminLogonValueInit, L"1") == 0)
                 {
                     bRequireLogonInit = FALSE;
                 }
```
This might make sense to avoid nested conditions.